### PR TITLE
Implemented confession view count

### PR DESCRIPTION
**Pull request by KAMALDEEN333**
The Confession view count feature has been created  which aims to track how many times each confession is viewed. This metric can later be leveraged for analytics and trending algorithms within the xconfess platform.

Added a view_count column to the confessions table.
Endpoint Modification:

On each GET /confessions/:id request, increment the view_count atomically.
View Filtering:

Prevented repeated views from the same IP or user within a 1-hour window to avoid inflated metrics.

Added basic tests to validate the increment logic and view filtering mechanism.

Ensured that the update to the view_count is atomic to prevent race conditions.
Used proper caching or rate-limiting techniques, if applicable, to track and filter repeated views from the same IP or user for the given time period.
Validated new database migrations to confirm that the view_count column integrates seamlessly without impacting any other features.
Wrote comprehensive tests covering normal view increments and edge cases for repeated views.

closes[#51]

## Summary by CodeRabbit

* **New Features**
  * Added a view count feature to confessions, displaying how many times each confession has been viewed.
  * View counts increment only for unique viewers within a one-hour window to prevent duplicate counts.
  * Introduced a caching mechanism to track recent views and optimize view count updates.
  * Added a new endpoint to retrieve confessions with updated view counts and error handling for missing confessions.
* **Tests**
  * Introduced new tests to verify correct view count behavior, uniqueness logic, and error handling.
* **Chores**
  * Updated and reorganized dependencies for improved clarity and correctness.

### CODE REVIEW

File: xconfess-backend/src/confession/confession.view-count.service.spec.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    expect(confession.view_count).toBe(1);
+  });
+
+  it('should not increment view count if viewed recently', async () => {
+    (repo.findOne as jest.Mock).mockResolvedValue({ id: '1', view_count: 5 });
+    (cache.hasViewedRecently as jest.Mock).mockResolvedValue(true);
+    const req = { user: { id: 'user1' }, ip: '127.0.0.1' };
+    const confession = await service.getConfessionByIdWithViewCount('1', req as any);
+    expect(confession.view_count).toBe(5);
+  });
```
[the remark this answers is not in the document]
Comment (KAMALDEEN333) [reply xconfess-backend/src/confession/confession.view-count.service.spec.ts]: resolved

File: xconfess-backend/data-source.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+import * as dotenv from 'dotenv';
+dotenv.config();
+
+export default new DataSource({
+  type: 'postgres',
+  host: process.env.DB_HOST,
+  port: Number(process.env.DB_PORT),
+  username: process.env.DB_USERNAME,
+  password: process.env.DB_PASSWORD,
+  database: process.env.DB_NAME,
```
[the remark this answers is not in the document]
Comment (KAMALDEEN333) [reply xconfess-backend/data-source.ts]: resolved

File: xconfess-backend/src/confession/confession.view-count.service.spec.ts [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+ });
+
+  it('should increment view count for new viewer', async () => {
+   const confession = { id: '1', view_count: 1 };
+    (service.getConfessionByIdWithViewCount as jest.Mock).mockResolvedValue(confession);
+    const req = { user: { id: 'user1' }, ip: '127.0.0.1' };
+    const result = await Controller.getConfessionById('1', req as any);
+    expect(result.view_count).toBe(1);
+   expect(service.getConfessionByIdWithViewCount).toHaveBeenCalledWith('1', req);
+  });
```
[the remark this answers is not in the document]
Comment (KAMALDEEN333) [reply xconfess-backend/src/confession/confession.view-count.service.spec.ts]: resolved
